### PR TITLE
pymol 2.3.0 (update)

### PR DESCRIPTION
**Pull request by YoshitakaMo**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----
This is a new formula of Open-source PyMOL ver. 2.3.0 released two days ago. MMTF format is now supported (`depends_on brewsci/bio/mmtf-cpp`).

### CODE REVIEW

**Comment by YoshitakaMo** [comment]
In build-linux, the report said in line 5852,
```
g++-5 -I/home/linuxbrew/.linuxbrew/opt/libxml2/include/libxml2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include/freetype2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include -std=c++11 -fPIC -D_PYMOL_LIBPNG -D_PYMOL_FREETYPE -D_PYMOL_INLINE -D_PYMOL_VMD_PLUGINS -D_HAVE_LIBXML -DMMTF_MSGPACK_USE_CPP11 -D_PYMOL_NUMPY -Icontrib/uiuc/plugins/include -Iov/src -Ilayer0 -Ilayer1 -Ilayer2 -Ilayer3 -Ilayer4 -Ilayer5 -Imodules/cealign/src -Ibuild/generated -Icontrib/uiuc/plugins/molfile_plugin/src -Icontrib/mmtf-c -I/home/linuxbrew/.linuxbrew/Cellar/pymol/2.3.0/libexec/lib/python3.7/site-packages/numpy/core/include -I/usr/include -I/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/openssl/include -I/home/linuxbrew/.linuxbrew/opt/sqlite/include -I/home/linuxbrew/.linuxbrew/opt/python/include/python3.7m -c contrib/uiuc/plugins/molfile_plugin/src/mapplugin.cpp -o build/temp.linux-x86_64-3.7/contrib/uiuc/plugins/molfile_plugin/src/mapplugin.o -Wno-narrowing -Wno-char-subscripts -ffast-math -funroll-loops -O3 -fcommon
g++-5 -I/home/linuxbrew/.linuxbrew/opt/libxml2/include/libxml2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include/freetype2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include -std=c++11 -fPIC -D_PYMOL_LIBPNG -D_PYMOL_FREETYPE -D_PYMOL_INLINE -D_PYMOL_VMD_PLUGINS -D_HAVE_LIBXML -DMMTF_MSGPACK_USE_CPP11 -D_PYMOL_NUMPY -Icontrib/uiuc/plugins/include -Iov/src -Ilayer0 -Ilayer1 -Ilayer2 -Ilayer3 -Ilayer4 -Ilayer5 -Imodules/cealign/src -Ibuild/generated -Icontrib/uiuc/plugins/molfile_plugin/src -Icontrib/mmtf-c -I/home/linuxbrew/.linuxbrew/Cellar/pymol/2.3.0/libexec/lib/python3.7/site-packages/numpy/core/include -I/usr/include -I/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/openssl/include -I/home/linuxbrew/.linuxbrew/opt/sqlite/include -I/home/linuxbrew/.linuxbrew/opt/python/include/python3.7m -c layer0/GraphicsUtil.cpp -o build/temp.linux-x86_64-3.7/layer0/GraphicsUtil.o -Wno-narrowing -Wno-char-subscripts -ffast-math -funroll-loops -O3 -fcommon
g++-5 -I/home/linuxbrew/.linuxbrew/opt/libxml2/include/libxml2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include/freetype2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include -std=c++11 -fPIC -D_PYMOL_LIBPNG -D_PYMOL_FREETYPE -D_PYMOL_INLINE -D_PYMOL_VMD_PLUGINS -D_HAVE_LIBXML -DMMTF_MSGPACK_USE_CPP11 -D_PYMOL_NUMPY -Icontrib/uiuc/plugins/include -Iov/src -Ilayer0 -Ilayer1 -Ilayer2 -Ilayer3 -Ilayer4 -Ilayer5 -Imodules/cealign/src -Ibuild/generated -Icontrib/uiuc/plugins/molfile_plugin/src -Icontrib/mmtf-c -I/home/linuxbrew/.linuxbrew/Cellar/pymol/2.3.0/libexec/lib/python3.7/site-packages/numpy/core/include -I/usr/include -I/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/openssl/include -I/home/linuxbrew/.linuxbrew/opt/sqlite/include -I/home/linuxbrew/.linuxbrew/opt/python/include/python3.7m -c layer0/File.cpp -o build/temp.linux-x86_64-3.7/layer0/File.o -Wno-narrowing -Wno-char-subscripts -ffast-math -funroll-loops -O3 -fcommon
g++-5 -I/home/linuxbrew/.linuxbrew/opt/libxml2/include/libxml2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include/freetype2 -I/home/linuxbrew/.linuxbrew/opt/freetype/include -std=c++11 -fPIC -D_PYMOL_LIBPNG -D_PYMOL_FREETYPE -D_PYMOL_INLINE -D_PYMOL_VMD_PLUGINS -D_HAVE_LIBXML -DMMTF_MSGPACK_USE_CPP11 -D_PYMOL_NUMPY -Icontrib/uiuc/plugins/include -Iov/src -Ilayer0 -Ilayer1 -Ilayer2 -Ilayer3 -Ilayer4 -Ilayer5 -Imodules/cealign/src -Ibuild/generated -Icontrib/uiuc/plugins/molfile_plugin/src -Icontrib/mmtf-c -I/home/linuxbrew/.linuxbrew/Cellar/pymol/2.3.0/libexec/lib/python3.7/site-packages/numpy/core/include -I/usr/include -I/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/openssl/include -I/home/linuxbrew/.linuxbrew/opt/sqlite/include -I/home/linuxbrew/.linuxbrew/opt/python/include/python3.7m -c layer1/PyMOLObject.cpp -o build/temp.linux-x86_64-3.7/layer1/PyMOLObject.o -Wno-narrowing -Wno-char-subscripts -ffast-math -funroll-loops -O3 -fcommon
g++-5: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
```
I don't have an idea to solve it. Please help me.

Moreover, in build-macos, the report just said
```
The specified paths did not match any files in /tmp
```
What does this message mean?

**Comment by sjackman** [comment]
`g++-5: internal compiler error: Killed (program cc1plus)`
This error means that the 4 GB of memory available on CircleCI was exceeded. Add right after `def install`…
```ruby
    # Reduce memory usage for CircleCI.
    ENV["MAKEFLAGS"] = "-j4" if ENV["CIRCLECI"]
```

**Comment by sjackman** [comment]
On macOS, expand the section immediately above `Persisting to Workspace`. The error `The specified paths did not match any files in /tmp` is related to this…
```
==> SKIPPING: brewsci/bio/pymol
XQuartz 2.7.11 (or newer) is required to install this formula. X11Requirement unsatisfied!
You can install with Homebrew Cask:
 brew cask install xquartz
You can download from:
 https://xquartz.macosforge.org
```
Uploading the bottle for macOS is failing because no bottle was built. I'll have to think of a workaround. Here's the buggy code:
https://github.com/brewsci/homebrew-bio/blob/4f9b7a1333ce7bf7edf62f4e021d24d73b526907/.circleci/config.yml#L75-L79

**Comment by YoshitakaMo** [comment]
Thank you for your suggestion. I added the option `ENV["MAKEFLAGS"] = "-j4" if ENV["CIRCLECI"]`, or tried `-j8`, but the problem was not solved yet.
I also hope another problem related to Xquartz will be solved.

**Comment by sjackman** [comment]
Try `ENV["MAKEFLAGS"] = "-j2" if ENV["CIRCLECI"]`

File: Formula/pymol.rb [hunk 1]
```diff
@@ -51,14 +47,8 @@ class Pymol < Formula
...
-    unless OS.mac?
-      # comment out glDebugMessageCallback if CentOS 7
-      # https://github.com/schrodinger/pymol-open-source/issues/2
-      inreplace "layer5/PyMOL.cpp",
-      "glDebugMessageCallback(gl_debug_proc, NULL);",
-      "// glDebugMessageCallback(gl_debug_proc, NULL); //"
-    end
-
+    # Reduce memory usage for CircleCI.
+    ENV["MAKEFLAGS"] = "-j8" if ENV["CIRCLECI"]
```
Comment (sjackman) [inline Formula/pymol.rb]: Please add a blank line after this line.

**Comment by sjackman** [comment]
Try this patch to fix macOS:
```diff
diff --git a/.circleci/config.yml b/.circleci/config.yml
index c381d7f..379bd9f 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -35,11 +35,11 @@ jobs:
             brew install patchelf
             brew tap linuxbrew/xorg
             brew test-bot --bintray-org=linuxbrew --root-url=https://linuxbrew.bintray.com/bottles-bio --git-name=LinuxbrewTestBot --git-email=testbot@linuxbrew.sh
+            mv brew-test-bot.xml brew-test-bot-linux.xml
       - persist_to_workspace:
           root: /tmp
           paths:
-            - bottles/*.json
-            - bottles/*.tar.gz
+            - bottles/*
       - store_test_results:
           path: /tmp/bottles
   build-macos:
@@ -72,11 +72,11 @@ jobs:
             mkdir /tmp/bottles
             cd /tmp/bottles
             brew test-bot --skip-setup --bintray-org=linuxbrew --root-url=https://linuxbrew.bintray.com/bottles-bio --git-name=LinuxbrewTestBot --git-email=testbot@linuxbrew.sh
+            mv brew-test-bot.xml brew-test-bot-macos.xml
       - persist_to_workspace:
           root: /tmp
           paths:
-            - bottles/*.json
-            - bottles/*.tar.gz
+            - bottles/*
       - store_test_results:
           path: /tmp/bottles
   collect-bottles:
```

**Comment by YoshitakaMo** [comment]
Thank you for your suggestion and patch. I tried again, but the ci/circleci: build-linux faied immediately,
```
fatal: --unshallow on a complete repository does not make sense
Exited with code 128
```
Homebrew-bio and build-macos seems OK.

**Comment by YoshitakaMo** [comment]
To circumvent the problem 'fatal: --unshallow on a complete repository does not make sense', I modified `config.yml` in line 14
```
git fetch origin --unshallow --tags || git fetch origin --depth=1000000 --tags
```
The test of ci/circleci: build-linux could run, but `g++-5: internal compiler error: Killed (program cc1plus)` message still appeared. I tried `-j2`, `-j4`, `-j8`, and `-j16` options with `ENV["MAKEFLAGS"]`, but seems no effect.

**Comment by sjackman** [comment]
Try `-j1`. It's possible that the build script (`setup.py`?) is ignoring the value of `MAKEFLAGS`. If so, we'll need to figure out what the correction option to pass to `setup.py` is to specify the number of concurrent compilers jobs.

**Comment by YoshitakaMo** [comment]
> It's possible that the build script (`setup.py`?) is ignoring the value of `MAKEFLAGS`.
That's it! I found a parallel build option `--jobs 1` when building with `setup.py`.
Now the problems have been solved. I am grateful for your kind advice.

File: Formula/pymol.rb [hunk 2]
```diff
@@ -81,9 +72,11 @@ def install
     args = %W[
       --install-scripts=#{libexec}/bin
       --install-lib=#{libexec}/lib/python#{xy}/site-packages
-      --pyqt PyQt5
+      --glut
+      --use-msgpackc c++11
     ]
     args << "--osx-frameworks" if OS.mac?
+    args << "--jobs 1" if ENV["CIRCLECI"] # Reduce memory usage for CircleCI.
```
Comment (sjackman) [inline Formula/pymol.rb]: Next time that you update this formula, could you please try larger values, say `--jobs=4`?

**Comment by sjackman** [comment]
Merged. Thanks again for your contribution to Brewsci/bio, @YoshitakaMo! 🏅

File: Formula/pymol.rb (hunk 2 above)
Comment (YoshitakaMo) [reply Formula/pymol.rb]: OK! I will do next time 😀

**Pull request closed without merging**